src/encoder.js

```diff
@@ -346,15 +340,36 @@ function encodeValue(value: any, target: Array<number | string>) {
   return target;
 }
 
-export default function encode(value: any): Array<number> {
+function findIndex(iterable, predicate) {
+  let index = 0;
+  for (let a of iterable) {
```

Comment:
This is going to depend on `Symbol.iterator` at runtime. We can't do that and remain IE 9 compatible.